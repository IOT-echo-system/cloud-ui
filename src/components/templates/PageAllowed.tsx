import React, {useEffect, useState} from 'react'
import {useSelector} from '../../hooks'
import {useRouter} from 'next/router'
import {Config} from '../../config'
import {Loader} from '../atoms'
import {PolicyUtils} from '../../utils/policyUtils'

type PageAllowedPropsType<T extends Record<string, unknown>> = {
  policy: string
  Component: React.ComponentType<T>
} & T

export const PageAllowed = <T extends Record<string, unknown>>(props: PageAllowedPropsType<T>): React.JSX.Element => {
  const {policy, Component} = props
  const {policies} = useSelector(state => state.user)
  const [validated, setValidated] = useState(false)
  const router = useRouter()

  useEffect(() => {
    const isAllowed = PolicyUtils.isValid(policies, policy)
    if (!isAllowed) {
      router.push(Config.HOME_PAGE_PATH).catch(() => ({}))
    }
    setValidated(true)
  }, [])

  if (!validated) {
    return <Loader loadingText={'Loading...'} />
  }

  return <Component {...props} />
}
