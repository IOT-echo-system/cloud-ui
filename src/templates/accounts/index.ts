export * from './signup/SignUp'
export * from './login/Login'
