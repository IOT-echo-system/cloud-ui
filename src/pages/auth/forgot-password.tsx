import type {NextPage} from 'next'
import React, {useState} from 'react'
import {ResetPassword, VerifyOtp} from '../../templates/accounts'
import {Config} from '../../config'

const ForgotPasswordPage: NextPage = () => {
  const [otpVerified, setOtpVerified] = useState(false)

  if (otpVerified) {
    return <ResetPassword title={'Change password'} withOldPassword={false} redirectTo={Config.LOGIN_PAGE_PATH} />
  }
  return <VerifyOtp title={'Forgot password'} setOtpVerified={setOtpVerified} />
}

export default ForgotPasswordPage
