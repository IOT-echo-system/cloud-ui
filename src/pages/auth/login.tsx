import type {NextPage} from 'next'
import React, {useEffect} from 'react'
import {LogIn} from '../../components/templates/auth'
import {AuthService} from '../../services'
import {Config} from '../../config'
import {useRouter} from 'next/router'

const LoginPage: NextPage = () => {
  const router = useRouter()
  useEffect(() => {
    AuthService.validate()
      .then(res => router.push(res.projectId ? Config.HOME_PAGE_PATH : Config.START_PAGE_PATH))
      .catch(() => ({}))
  }, [])

  return <LogIn />
}

export default LoginPage
