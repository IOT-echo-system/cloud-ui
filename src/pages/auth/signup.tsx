import type {NextPage} from 'next'
import React from 'react'
import {SignUp} from '../../components/templates/auth'

const SignUpPage: NextPage = () => {
  return <SignUp />
}

export default SignUpPage
