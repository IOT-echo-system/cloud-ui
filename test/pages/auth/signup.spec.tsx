import {render} from '@testing-library/react'
import SignUpPage from '../../../src/pages/auth/signup'
import {SignUp} from '../../../src/templates/auth'

jest.mock('../../../src/templates/accounts/signup/SignUp')

describe('SignUp page test', () => {
  beforeEach(jest.clearAllMocks)
  afterEach(jest.clearAllMocks)

  it('should match with the snapshot', () => {
    const {container} = render(<SignUpPage />)

    expect(container).toMatchSnapshot()
    expect(SignUp).toHaveBeenCalledTimes(1)
    expect(SignUp).toHaveBeenCalledWith({}, {})
  })
})
