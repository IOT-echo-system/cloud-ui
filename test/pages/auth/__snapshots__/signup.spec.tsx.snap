// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp page test should match with the snapshot 1`] = `<div />`;
